Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -61,29 +67,43 @@ void DataReaderTask::run(ProcessingContext& pc)
   auto device = pc.services().get<o2::framework::RawDeviceService>().device();
   auto outputRoutes = pc.services().get<o2::framework::RawDeviceService>().spec().outputs;
   auto fairMQChannel = outputRoutes.at(0).channel;
-  int inputcount = 0;
+  mDataSpec = o2::header::gDataDescriptionRawData;
+
+  std::vector<InputSpec> dummy{InputSpec{"dummy", ConcreteDataMatcher{"TRD", mDataSpec, 0xDEADBEEF}}};
+  // if we see requested data type input with 0xDEADBEEF subspec and 0 payload this means that the "delayed message"
+  //   // mechanism created it in absence of real data from upstream. Processor should send empty output to not block the workflow
+
+  for (const auto& ref : InputRecordWalker(pc.inputs(), dummy)) {
+    const auto dh = o2::framework::DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+    if (dh->payloadSize == 16 || dh->payloadSize == 0) {
+      LOGP(WARNING, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : assuming no payload for all links in this TF",
+           dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      sendData(pc, true); //send the empty tf data.
+      return;
+    }
+    LOG(info) << " matched DEADBEEF";
+  }
+  //TODO combine the previous and subsequent loops.
   /* loop over inputs routes */
   for (auto iit = pc.inputs().begin(), iend = pc.inputs().end(); iit != iend; ++iit) {
     if (!iit.isValid()) {
       continue;
     }
     /* loop over input parts */
+    int inputpartscount = 0;
     for (auto const& ref : iit) {
-
+      if (mVerbose) {
+        const auto dh = DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+        LOGP(info, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : assuming no payload for all links in this TF",
+             dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      }
```

Comment:
still here, please remove this

Reply:
but that is inside the mVerbose, its what I asked about the previous time, as there was a log info in the line previous to this

Reply:
Not sure which question you refer to, but here you are supposed to process a healthy data, which with `mVerbose` you would print `assuming no payload for all links ...`  ?

Reply:
bad paste, removing the text after the :